### ISSUE TITLE
Update Waterfall layout logic

### ISSUE THREAD

**Issue by LFDanLu** [adobe/react-spectrum#2282]
# 🐛 Bug Report 

Waterfall layout should support a min/max item height so that there aren't extreme differences in card heights due to large variances in the images they show.

Other things like scale support will be covered in a separate issue (waiting on design)
- Technically scale support is already in Waterfall layout logic since it will measure the card's true height on the second pass when calculating the height
